inc/Ui/Search.php

```diff
@@ -499,7 +499,7 @@ protected function getSearchIntroHTML($query)
     public function createPagenameFromQuery($parsedQuery)
     {
         $cleanedQuery = cleanID($parsedQuery['query']);
-        if ($cleanedQuery === $parsedQuery['query']) {
+        if (strtolower($cleanedQuery) === strtolower($parsedQuery['query'])) {
```

Comment:
1. As I understand it, [`cleanID()` makes strings lowercase anyway][0], so that first `strtolower()` is unnecessary?
2. `cleanID()` doesn't use `strtolower()`, but [`utf8_strtolower()`][1]. So your method might break if one uses non-english letters and a not the corresponding locale. Hence, I suggest switching your second `strtolower()` to `utf8_strtolower()` as well.
I haven't tested it locally, but I would suggest also adding another testcase that uses both German capital umlaute and some Russian capital letters.


[0]: https://github.com/splitbrain/dokuwiki/blob/1fb6cf136354a75a3ab6a6294769b789928459ac/inc/pageutils.php#L128
[1]: https://github.com/splitbrain/dokuwiki/blob/1fb6cf136354a75a3ab6a6294769b789928459ac/inc/utf8.php#L366

Reply:
Thanks for pointing it out, `utf8_strtolower` is denifitely a good idea.